keanu-project/src/main/java/io/improbable/keanu/vertices/dbltensor/SimpleDoubleTensor.java

```diff
@@ -0,0 +1,319 @@
+        value = value - that;
+        return this;
+    }
+
+    @Override
+    public DoubleTensor plusInPlace(double that) {
+        value = value + that;
+        return this;
+    }
+
+    @Override
+    public DoubleTensor timesInPlace(double that) {
+        value = value * that;
+        return this;
+    }
+
+    @Override
+    public DoubleTensor divInPlace(double that) {
+        value = value / that;
+        return this;
+    }
+
+    @Override
+    public DoubleTensor powInPlace(DoubleTensor exponent) {
+        if (exponent.isScalar()) {
+            value = Math.pow(value, exponent.scalar());
+        } else {
+            throw new IllegalArgumentException("Only scalar tensors supported");
+        }
+
+        return this;
+    }
+
+    @Override
+    public DoubleTensor powInPlace(double exponent) {
+        value = Math.pow(value, exponent);
+        return this;
+    }
+
+    @Override
+    public DoubleTensor logInPlace() {
+        value = Math.log(value);
+        return this;
+    }
+
+    @Override
+    public DoubleTensor sinInPlace() {
+        value = Math.sin(value);
+        return this;
+    }
+
+    @Override
+    public DoubleTensor cosInPlace() {
+        value = Math.cos(value);
+        return this;
+    }
+
+    @Override
+    public DoubleTensor asinInPlace() {
+        value = Math.asin(value);
+        return this;
+    }
+
+    @Override
+    public DoubleTensor acosInPlace() {
+        value = Math.acos(value);
+        return this;
+    }
+
+    @Override
+    public DoubleTensor expInPlace() {
+        value = Math.exp(value);
+        return this;
+    }
+
+    @Override
+    public DoubleTensor minusInPlace(DoubleTensor that) {
+        if (that.isScalar()) {
+            minusInPlace(that.scalar());
+        } else {
+            throw new IllegalArgumentException("Only scalar tensors supported");
+        }
+        return this;
+    }
+
+    @Override
+    public DoubleTensor plusInPlace(DoubleTensor that) {
+        if (that.isScalar()) {
+            plusInPlace(that.scalar());
+        } else {
+            throw new IllegalArgumentException("Only scalar tensors supported");
+        }
+        return this;
+    }
+
+    @Override
+    public DoubleTensor timesInPlace(DoubleTensor that) {
+        if (that.isScalar()) {
+            timesInPlace(that.scalar());
+        } else {
+            throw new IllegalArgumentException("Only scalar tensors supported");
+        }
+        return this;
+    }
+
+    @Override
+    public DoubleTensor divInPlace(DoubleTensor that) {
+        if (that.isScalar()) {
+            divInPlace(that.scalar());
+        } else {
+            throw new IllegalArgumentException("Only scalar tensors supported");
+        }
+        return this;
+    }
+
+    @Override
+    public DoubleTensor unaryMinusInPlace() {
+        value = -value;
+        return this;
+    }
+
+    @Override
+    public double[] getLinearView() {
```

Comment:
I find `linearView` quite a strange way of expressing this flat concept. Is that what nd4j call it?

Reply:
Linear view is a 1d view of the underlying buffer in Nd4j, which is not what this actually is here. I've changed it to get getFlattenedView() and it returns an actual view (meaning editing it will actually edit the underlying tensor).